articles/customization-using-bulma/index.md

```diff
@@ -0,0 +1,233 @@
+
+### Getting Started
+
+Let us add Bulma to our web application project. We can do this by one of the following ways.
+
+- We can use the Bulma CDN to add Bulma to our webpage. In the website `<head>`, add the following line.
+
+```html
+<link
+  rel="stylesheet"
+  href="https://cdn.jsdelivr.net/npm/bulma@0.9.0/css/bulma.min.css"
+/>
+```
+
+- Download the npm package from the npm repository.
+
+```
+npm install bulma
+```
+
+- We can also download the `bulma.css` file from [here](https://github.com/jgthms/bulma/tree/master/css) and include it in our webpage using the `<link>` tag.
+
+### Column System
+
+One of the best features of Bulma is the responsive column grid system. For building a columns layout,
+
+- Every row in Bulma is defined by a `columns` container.
+- Every column should be defined as a `column` and should be nested within the `columns` container.
+
+For example,
+
+```html
+<div class="columns">
+  <div class="column">1st column</div>
+  <div class="column">2nd column</div>
+  <div class="column">3rd column</div>
+</div>
+```
+
+This snippet would create a responsive column layout with 3 columns of equal width. You can also specify the column size by using the size classes.
+
+```html
+<div class="columns">
+  <div class="column is-half">1st column</div>
+  <div class="column is-one-quarter">2nd column</div>
+  <div class="column">3rd column</div>
+</div>
+```
+
+The above snippet would create three columns. One column occupies half the space, and the other two occupy a quarter of the space. For more options on columns, visit [this link](https://bulma.io/documentation/columns/).
+
+### Components
+
+Bulma comes with many in-built UI components. Let us try to create a card component.
+
+```html
+<div class="card">
+  <div class="card-content">
+    <p class="title">
+      Lorem ipsum dolor, sit amet consectetur adipisicing elit. Architecto,
+      corrupti.
+    </p>
+    <p class="subtitle">Lorem Ipsum.</p>
+  </div>
+  <footer class="card-footer">
+    <p class="card-footer-item">
+      <span>Footer-Left</span>
+    </p>
+    <p class="card-footer-item">
+      <span>Footer-Right</span>
+    </p>
+  </footer>
+</div>
+```
+
+This code snippet would create a card component that would look like this:
+
+![Bulma Card](/engineering-education/customization-using-bulma/card.png)
+
+Bulma has many other components, for which the documentation can be viewed [here](https://bulma.io/documentation/components/).
+
+### Customizing Variables
+
+For customizing Bulma, there are four kinds or levels of variables:
+
+- Initial Variables -- There are global variables with literal values.
+- Derived Variables -- These are derived from other variables.
+- Generic Variables -- HTML elements that cary no CSS class.
+- Element Variables -- variables that are specific to a bulma element.
+
+Now, for customizing these Bulma Variables, let us set up sass and bulma.
```

Comment:
Consider adding a brief explanation of Sass. Readers who are new to CSS frameworks most likely don't know what Sass is.